# Andrey Tomarovsky

_Software enthusiast in bioinformatics based in St. Petersburg, Russia_ <br>

üìß[Email](mailto:andrey.tomarovsky@gmail.com) / üí¨[Telegram](https://t.me/tomatoma) / üí¨[Facebook](https://facebook.com/andrey.tomorovsky) / üß¨[Rosalind](http://rosalind.info/users/toma/) / üìú[Google Schular](https://scholar.google.com/citations?user=2hnNX4sAAAAJ)

## üë®üèª‚Äçüéì Education

**Bachelor's degree** in Biotechnologies<br>
[Belgorod State National Research University](https://www.bsu.edu.ru/bsu/) - Belgorod, Russia _(2015 - 2019)_

**Master's degree** in Bioinformatics<br>
[St. Petersburg State University](https://spbu.ru/) - St. Petersburg, Russia _(2019 - 2021)_

**Ph.D.** in Biology<br>
[Novosibirsk State University](https://www.nsu.ru/n/) - Novosibirsk, Russia _(2021 - now)_

## üèÜ Accomplishments

ResOps experience on computing clusters [MSU FBB](https://makarich.fbb.msu.ru/), [IMCB](https://www.mcb.nsc.ru/) and [ITMO](https://itmo.ru/ru/).
[JetBrains Research](https://research.jetbrains.org/) Scholarship (2020).

### üõ† Courses

**[Coursera:](https://www.coursera.org/user/9470ce667a3222f90f4566e23282df3b)**
  - Capstone: Retrieving, Processing, and Visualizing Data with Python
  - Programming for Everybody (Getting Started with Python)
  - Python Data Structures
  - –ó–Ω–∞–∫–æ–º—Å—Ç–≤–æ —Å R –∏ –±–∞–∑–æ–≤–∞—è —Å—Ç–∞—Ç–∏—Å—Ç–∏–∫–∞

**[Stepik.org:](https://stepik.org/users/73031805)**
  - –í–≤–µ–¥–µ–Ω–∏–µ –≤ Linux
  - –ü—Ä–æ–≥—Ä–∞–º–º–∏—Ä–æ–≤–∞–Ω–∏–µ –Ω–∞ Python
  - Python: –æ—Å–Ω–æ–≤—ã –∏ –ø—Ä–∏–º–µ–Ω–µ–Ω–∏–µ
  - –ë–∏–æ—Ç–µ—Ö–Ω–æ–ª–æ–≥–∏–∏: –≥–µ–Ω–Ω–∞—è –∏–Ω–∂–µ–Ω–µ—Ä–∏—è
  - –°–µ–∫–≤–µ–Ω–∏—Ä–æ–≤–∞–Ω–∏–µ 3-–≥–æ –ø–æ–∫–æ–ª–µ–Ω–∏—è –Ω–∞ Oxford Nanopore
  - –í–≤–µ–¥–µ–Ω–∏–µ –≤ –º–æ–ª–µ–∫—É–ª—è—Ä–Ω—É—é –±–∏–æ–ª–æ–≥–∏—é –∏ –±–∏–æ–º–µ–¥–∏—Ü–∏–Ω—É
  - –ú–æ–ª–µ–∫—É–ª—è—Ä–Ω–∞—è –±–∏–æ–ª–æ–≥–∏—è –∏ –≥–µ–Ω–µ—Ç–∏–∫–∞
  - –ì–µ–Ω–µ—Ç–∏–∫–∞ –∏ –≥–µ–Ω–æ–º–∏–∫–∞ –ø–æ–ø—É–ª—è—Ü–∏–π
  - –ú–æ–ª–µ–∫—É–ª—è—Ä–Ω–∞—è —Ñ–∏–ª–æ–≥–µ–Ω–µ—Ç–∏–∫–∞
  - Python Skills for Handling Biological Data
  - –ê–ª–≥–æ—Ä–∏—Ç–º—ã: —Ç–µ–æ—Ä–∏—è –∏ –ø—Ä–∞–∫—Ç–∏–∫–∞. –°—Ç—Ä—É–∫—Ç—É—Ä—ã –¥–∞–Ω–Ω—ã—Ö
  - –û—Å–Ω–æ–≤—ã Git

**[Sololearn.com:](https://www.sololearn.com/Profile/10277208/?ref=app)**
  - Python 3 Tutorial course
  - SQL Fundamentals course
  - C++ Tutorial course
  - Git

## üìå On The Side

**Script package for Bioinformatics** @ [Biocrutch](https://github.com/tomarovsky/Biocrutch) _(Aug 2020 - Present)_<br>
A project containing conceptual scripts for bioinformatics.
  - [SRA_toolkit.py](https://github.com/tomarovsky/Biocrutch/blob/master/scripts/SRA_toolkit.py). The program parses the link from the Sequence Read Archive (SRA) and allows you to download reads in the sra format. The program also checks the integrity of the finished reads by parsing the required metrics of the source files.
  - [pseudoautosomal_region.py](https://github.com/tomarovsky/Biocrutch/blob/master/scripts/genomecov/pseudoautosomal_region.py). A script for determining the coordinates of the pseudo-autosomal region on the sex chromosome. The output is a BED file with the coordinates of the pseudoautosomal region.
  - [RepeatMasking scripts](https://github.com/tomarovsky/Biocrutch/tree/master/scripts/RepeatMasking). Scripts for converting TRF, RepeatMasker and WindowMasker output to GFF format.
  - [ema_bin_to_fastq.py](https://github.com/tomarovsky/Biocrutch/blob/master/scripts/10x/ema_bin_to_fastq.py). Combines Ema output files into reverse, forward and barcode-only file.
  - [quast_core.py](https://github.com/tomarovsky/Biocrutch/blob/master/scripts/quast_core.py). The program is an alternative to the publicly available Quast program. Its main differences are: 
    1. adding only the necessary cutoffs.
    2. counting missing N values.
    3. the output of the program is a pandas dataframe used for further analysis.
    4. output to a convenient csv file format. And others.
  - [coverage_statistics.py](https://github.com/tomarovsky/Biocrutch/blob/master/scripts/genomecov/coverage_statistics.py). Script for calculating median, average, maximum and minimum coverage.
    1. calculate stats for whole genome.
    2. calculate stats for each scaffold.
    3. calculate stats stacking windows.
    Header-less tab-separated input file with 3 columns: scaffold_id, position(1-based), coverage. Script works with the output of Bedtools Genomecov and Mosdepth programs. 
  <br><br>

## üìù Articles

  - Totikov, Azamat, Andrey Tomarovsky, Dmitry Prokopov, Aliya Yakupova, Tatiana Bulyonkova, Lorena Derezanin, Dmitry Rasskazov, –∏ –¥—Ä. ¬´Chromosome-Level Genome Assemblies Expand Capabilities of Genomics for Conservation Biology¬ª. Genes 12, –≤—ã–ø. 9 (—Å–µ–Ω—Ç—è–±—Ä—å 2021 –≥.): 1336. https://doi.org/10.3390/genes12091336.
  - Totikov, A.; Tomarovsky, A.; Derezanin, L.; Dudchenko, O.; Lieberman-Aiden, E.; Koepfli, K.; Kliver, S. Chromosome-Level Genome Assemblies: Expanded Capabilities for Conservation Biology Research. Proceedings 2021, 76, 10. doi: 10.3390/IECGE-07149
  - Tomarovsky A.A., Ryadinskaya O.P., Toward an ecological classification of soil bacteria. We make the future: Collected papers of students / Editor-in-Chief: I.V. Borisovskaya, E.N. Taranova. ‚Äì V. 1. ‚Äì Belgorod, 2018. ‚Äì 204 —Ä.

## üë®üèª‚Äçüíº Conferences

  - Tomarovsky, A.; Azamat A. Totikov; Violetta R. Beklemisheva; Polina L. Perelman; Natalia A. Serdyokova; Tatiana Bulyonkova; Ksenia A. Koniaeva; Alexei V. Abramov; Alexander S. Graphodatsky; Klaus-Peter Koepfli; Roger A. Powell; Sergei F. Kliver, Assembly and annotation of sable (Martes zibellina) and pine marten (Martes martes) genomes, Moscow Conference on Computational Molecular Biology (MCCMB), July 30th ‚Äî August 2nd, 2021.
  - Azamat A. Totikov; Andrey A. Tomarovsky; Polina Perelman; Natalia Serdyokova; Violetta R. Beklemisheva; Tatiana M. Bulyonkova; In√™s Miranda; Liliana Farelo; Karol Zub; Viktor V. Panov; Anna S. Mukhacheva; Alexei V. Abramov; Klaus-Peter Koepfli; Alexander S. Graphodatsky; Jose Melo-Ferreira; Sergei F. Kliver, Reconstruction of the demographic history for three populations of the least weasel Mustela nivalis, Moscow Conference on Computational Molecular Biology (MCCMB), July 30th ‚Äî August 2nd, 2021.
  - Totikov, A., Tomarovsky, A., Derezanin, L., Dudchenko, O., Lieberman-Aiden, E., Koepfli, K., & Kliver, S. (2020, November). Chromosome-Level Genome Assemblies Expanded Capabilities of Conservation Biology. In Presented at the 1st International Electronic Conference on Genes: Theoretical and Applied Genomics (Vol. 2, p. 30).
  - –¢–æ—Ç–∏–∫–æ–≤ –ê.–ê., –¢–æ–º–∞—Ä–æ–≤—Å–∫–∏–π –ê.–ê., –ê–Ω–∞–ª–∏–∑ –∫–ª–∞—Å—Ç–µ—Ä–æ–≤ –≥–µ–Ω–æ–≤ –±–∏–æ—Å–∏–Ω—Ç–µ–∑–∞ –º–µ—Ç–∞–±–æ–ª–∏—Ç–æ–≤ –≤ –±–∞–∫—Ç–µ—Ä–∏–∞–ª—å–Ω—ã—Ö –≥–µ–Ω–æ–º–∞—Ö. 24-—è –ú–µ–∂–¥—É–Ω–∞—Ä–æ–¥–Ω–∞—è –ü—É—â–∏–Ω—Å–∫–∞—è —à–∫–æ–ª–∞–∫–æ–Ω—Ñ–µ—Ä–µ–Ω—Ü–∏—è –º–æ–ª–æ–¥—ã—Ö —É—á–µ–Ω—ã—Ö. 2020, –ü—É—â–∏–Ω–æ. –°–±–æ—Ä–Ω–∏–∫ —Ç–µ–∑–∏—Å–æ–≤, 2020. ‚Äì 446 —Å. ISBN 978-5-91874-901-2.
  - –¢–æ–º–∞—Ä–æ–≤—Å–∫–∏–π –ê.–ê., –õ–∏—Ö–∏–¥—á–µ–Ω–∫–æ –í.–ò., –ü–æ–ª–∏–≤–∞–Ω–æ–≤–∞ –ï.–Æ., –°–µ—Ä–∏–∫–æ–≤–∞ –ù.–í., –ê–Ω–∞–ª–∏–∑ –∏ —Å—Ä–∞–≤–Ω–µ–Ω–∏–µ —Ä–µ–∑—É–ª—å—Ç–∞—Ç–æ–≤ –ø–æ—Å–ª–µ –ø—Ä–∏–º–µ–Ω–µ–Ω–∏—è —Ñ–µ—Ä–º–µ–Ω—Ç–∞ Alphalase AFP –Ω–∞ –∫–ª–µ—Ç–∫–∏ Corynebacterium glutamicum. CXIX –ú–µ–∂–¥—É–Ω–∞—Ä–æ–¥–Ω–∞—è –Ω–∞—É—á–Ω–æ–ø—Ä–∞–∫—Ç–∏—á–µ—Å–∫–∞—è –∫–æ–Ω—Ñ–µ—Ä–µ–Ω—Ü–∏—è ¬´–ú–æ–ª–æ–¥–æ–π –∏—Å—Å–ª–µ–¥–æ–≤–∞—Ç–µ–ª—å: –≤—ã–∑–æ–≤—ã –∏ –ø–µ—Ä—Å–ø–µ–∫—Ç–∏–≤—ã¬ª, 05/2019.
  - –¢–æ–º–∞—Ä–æ–≤—Å–∫–∏–π –ê.–ê., –õ–∏—Ö–∏–¥—á–µ–Ω–∫–æ –í.–ò., –ü–æ–ª–∏–≤–∞–Ω–æ–≤–∞ –ï.–Æ., –°–µ—Ä–∏–∫–æ–≤–∞ –ù.–í., –î–µ–π—Å—Ç–≤–∏–µ —Ñ–µ—Ä–º–µ–Ω—Ç–∞ Proteinase T –Ω–∞ –∫–ª–µ—Ç–∫–∏ –ø—Ä–æ–∏–∑–≤–æ–¥—Å—Ç–≤–µ–Ω–Ω–æ–≥–æ —à—Ç–∞–º–º–∞ Corynebacterium glutamicum. XIX –í—Å–µ—Ä–æ—Å—Å–∏–π—Å–∫–∞—è –∫–æ–Ω—Ñ–µ—Ä–µ–Ω—Ü–∏—è –º–æ–ª–æ–¥—ã—Ö —É—á–µ–Ω—ã—Ö ¬´–ë–∏–æ—Ç–µ—Ö–Ω–æ–ª–æ–≥–∏—è –≤ —Ä–∞—Å—Ç–µ–Ω–∏–µ–≤–æ–¥—Å—Ç–≤–µ, –∂–∏–≤–æ—Ç–Ω–æ–≤–æ–¥—Å—Ç–≤–µ –∏ —Å–µ–ª—å—Å–∫–æ—Ö–æ–∑—è–π—Å—Ç–≤–µ–Ω–Ω–æ–π –º–∏–∫—Ä–æ–±–∏–æ–ª–æ–≥–∏–∏¬ª –°–±–æ—Ä–Ω–∏–∫ —Ç–µ–∑–∏—Å–æ–≤ –¥–æ–∫–ª–∞–¥–æ–≤, –≥. –ú–æ—Å–∫–≤–∞, 04/2019;
  - –õ–∏—Ö–∏–¥—á–µ–Ω–∫–æ –í.–ò., –¢–æ–º–∞—Ä–æ–≤—Å–∫–∏–π –ê.–ê., –ü–æ–ª–∏–≤–∞–Ω–æ–≤–∞ –ï.–Æ., –°–µ—Ä–∏–∫–æ–≤–∞ –ù.–í., –í–ª–∏—è–Ω–∏–µ –º–µ—Ç–∞–±–∏—Å—É–ª—å—Ñ–∏—Ç–∞ –Ω–∞—Ç—Ä–∏—è (223) –Ω–∞ —Å–æ—Ö—Ä–∞–Ω–Ω–æ—Å—Ç—å –ª–∏–∑–∏–Ω–∞ –≤ –ø—Ä–æ—Ü–µ—Å—Å–µ —Ö—Ä–∞–Ω–µ–Ω–∏—è. XIX –í—Å–µ—Ä–æ—Å—Å–∏–π—Å–∫–∞—è –∫–æ–Ω—Ñ–µ—Ä–µ–Ω—Ü–∏—è –º–æ–ª–æ–¥—ã—Ö —É—á–µ–Ω—ã—Ö ¬´–ë–∏–æ—Ç–µ—Ö–Ω–æ–ª–æ–≥–∏—è –≤ —Ä–∞—Å—Ç–µ–Ω–∏–µ–≤–æ–¥—Å—Ç–≤–µ, –∂–∏–≤–æ—Ç–Ω–æ–≤–æ–¥—Å—Ç–≤–µ –∏ —Å–µ–ª—å—Å–∫–æ—Ö–æ–∑—è–π—Å—Ç–≤–µ–Ω–Ω–æ–π –º–∏–∫—Ä–æ–±–∏–æ–ª–æ–≥–∏–∏¬ª –°–±–æ—Ä–Ω–∏–∫ —Ç–µ–∑–∏—Å–æ–≤ –¥–æ–∫–ª–∞–¥–æ–≤, –≥. –ú–æ—Å–∫–≤–∞, 04/2019;

## üí¨ Languages

**Russian**: Native <br>
**English**: Upper-Intermediate
<br><br>


