# Andrey Tomarovsky

_Software enthusiast in bioinformatics based in St. Petersburg, Russia_ <br>

üìß[Email](mailto:andrey.tomarovsky@gmail.com) / üí¨[Telegram](https://t.me/tomatoma) / üí¨[Facebook](https://facebook.com/andrey.tomorovsky) / üìú[Google Schular](https://scholar.google.com/citations?user=2hnNX4sAAAAJ)

üìù[PDF CV Download](https://github.com/tomarovsky/tomarovsky.github.io/raw/master/cv.pdf)

## üë®üèª‚Äçüéì Education

| _September, 2021 - present_ | **PhD student in Genetics**. Novosibirsk State University, Novosibirsk, Russia. _PhD thesis_: "Obtaining genomic assemblies and phylogenetic analysis of members of the genus Martes (fam. Mustelidae)". |
| _September, 2019 - July 2021_ | **MS in Bioinformatics**. Saint-Petersburg State University, St. Petersburg, Russia. _MS thesis_: "Assembly and annotation of the sable (Martes zibellina) and pine marten (Martes martes) genomes". |
| _September, 2015 - July 2019_ | **BS in Biotechnologies**. Belgorod State National Research University, Belgorod, Russia. |

## üèÜ Work experience

1) _March, 2021 - present_ **Research programmer at the Genomic Diversity Research Center, ITMO University**. Conducts research on genomics of the genus Martes:
- Quality control and filtering of sequencing data.
- _De novo_ assembly and quality control.
- Genome annotation:
  - Whole genome alignments (pairwise and multiple) and coverage statistics.
  - Calculating the coordinates of the pseudoautosomal region (custom algorithm was developed).
  - _De novo_ assembly and annotation of genome repeats.
  - Protein-coding and non-coding genes annotation.
- Calling and filtering of genetic variants.
- Visualization and analysis of heterozygosity.
- Phylogenetic and evolutionary analysis.
- Comparison and analysis of the results obtained.
- Pipeline development based on Bash and Snakemake.

2) _July, 2021 - present_ **Technical and software support for [Blastim](https://agency.blastim.ru/educenter) courses**:

- Python and Linux for Bioinformatics and Biology (July 2021).
- Analysis of NGS data (September 2021, February 2022).

3) _December, 2020 - present_ **ResOps and system administration experience on computing cluster MSU FBB**. 
- Assistance in writing commands and pipelines for bioinformatic analyses.
- Software support for cluster users.
- Creating a cluster [documentation](https://ma.fbb.msu.ru/faq/).

### üõ† Skills

| **OS**: | Linux, Windows |
| **Shell**: | Bash. <br/> A good knowledge of the various shell tools, such as Awk, Grep, Sed |
| **Programming**: | Python |
| **Python libraries**: | Biopython, Matplotlib, Numpy, Pandas, Scikit-learn <br/>Experience in writing various scripts and data visualizations in Jupyter Notebook and individual Python packages. <br/> - Parsing data from files or websites to Pandas dataframes. <br/> - Calculation of average, median, minimum and maximum values in datasets. <br/> - Visualization of results using Matplotlib in the form of plots, histograms, Venn diagrams. <br/> - A little experience in ML (kNN, clustering, linear regression) |
| **Statistics**: | R |
| **R libraries**: | readxl, dplyr, car, cowplot, ggplot2 <br/> Experience analyzing various datasets, such as those containing information on different types of cancer and patient survival times. <br/> - Linear and multiple regression. <br/> - Description and significance testing of linear models. <br/> - Comparison of linear models. <br/> - Testing statistical hypotheses. |
| **Workflow managers**: | Snakemake <br/> Experience in writing complex Snakemake pipelines including benchmarking, logging, task grouping and running on a compute cluster. There is experience in collaborative development. |
| **Workload managers**: | Slurm, PBS. <br/> ResOps experience on computing clusters MSU FBB, ICG, IMCB and ITMO. <br/> - Running large-scale computational tasks using Slurm, PBS and Snakemake. <br/> - Installation and interaction with Conda environments. |
| **Others**: | - SQL (creating a database, simple and medium complexity queries) <br/> - Circos (basic level, experience in visualization of mDNA and its coverage) <br/> - Tcl (basic level, experience in writing module files) |

## üìå On The Side

_Snakemake pipelines:_

| **[ITSpipe](https://github.com/tomarovsky/ITSpipe)**. Pipeline for the analysis of ITS sequences from the ribosomal cluster. Coverage visualization using Matplotlib and variant calling using Gatk, Pisces, and Bcftool is performed. |
| **[BuscoPhylo](https://github.com/mahajrod/BuscoPhylo)**. Pipeline to construct species phylogenies using universal single-copy orthologs BUSCOs. |
| **[varcaller](https://github.com/rutaolta/varcaller)**. Pipeline for calling genetic variants correctly. Includes visualization of coverage and calculation of PAR coordinates. |

_Others:_

| **[Biocrutch](https://github.com/tomarovsky/Biocrutch)**. A custom python package for bioinformatics research. My project contains bioinformatics scripts for genome and coverage statistics, repeats masking, determining coordinates of pseudoautosomal region, filtering 10XGenomics linked reads, PSMC date combine and others. |
| **[Bashare](https://github.com/tomarovsky/bashare)**. The repository contains custom Bash scripts and pipelines for data processing. |

## üìù Grants

- Russian Foundation for Basic Research, grant ‚Ññ 20-04-00808 A, ‚ÄúGenomes and genetic diversity of mustelids (fam. Mustelidae) of Russia and South-Eastern Asia‚Äù.

## üìù Articles

  - (submitted) Dere≈æanin, L.; Bla≈æytƒó, A.; Dobrynin, P.; Duch√™ne, D.A.; Grau, J.H.; Hofreiter, M.; Jeon, S.; Kliver, S.; Koepfli, K.P.; Meneghini, D.; Preick, M.; Tomarovsky, A; Totikov, A.; Fickel, J.; F√∂rster, D.W. Genome assembly of the tayra (Eira barbara, Mustelidae) and comparative genomic analysis reveal adaptive genetic variation in the subfamily Guloninae. 2021.
  - Totikov, A.; Tomarovsky, A.; Prokopov, D.; Yakupova, A.; Bulyonkova, T.; Derezanin, L.; Rasskazov, D.; Wolfsberger, W.; Koepfli, K.P.; Oleksyk, T.K. and Kliver, S. Chromosome-Level Genome Assemblies Expand Capabilities for Conservation Biology. Genes 2021, 12, x. https://doi.org/10.3390/genes12091336
  - Tomarovsky A.A., Ryadinskaya O.P., Toward an ecological classification of soil bacteria. We make the future: Collected papers of students / Editor-in-Chief: I.V. Borisovskaya, E.N. Taranova. ‚Äì V. 1. ‚Äì Belgorod, 2018. ‚Äì 204 —Ä.

## üë®üèª‚Äçüíº Conferences

- Tomarovsky, A.; Totikov, A.; Beklemisheva, V.; Perelman, P.; Serdyokova, N.; Bulyonkova, T.; Koniaeva, K.; Abramov, A.; Graphodatsky, A.; Koepfli K.; Powell R.; Kliver S. Assembly and annotation of the sable (Martes zibellina) and pine marten (Martes martes) genomes. ISBN: 978-5-901158-32-6.
- Totikov, A.; Tomarovsky, A.; Perelman, P.; Serdyokova, N.; Beklemisheva, V.; Bulyonkova, T.; Zub, K.; Panov, V.; Mukhacheva, A.; Abramov, A.; Koepfli, K.; Graphodatsky, A.; Melo-Ferreira, J.; Kliver, S. Reconstruction of the demographic history for three populations of the least weasel Mustela nivalis. ISBN: 978-5-901158-32-6.
- Totikov, A.; Tomarovsky, A.; Perelman, P.; Serdyokova, N.; Beklemisheva, V.; Bulyonkova, T.; Zub, K.; Panov, V.; Mukhacheva, A.; Abramov, A.; Koepfli, K.; Graphodatsky, A.; Melo-Ferreira, J.; Kliver, S. Reconstruction of the demographic history for three populations of the least weasel Mustela nivalis. ISBN: 978-5-901158-32-6.
- Dere≈æanin, L.; Bla≈æytƒó, A.; Dobrynin, P.; Duch√™ne, D.A.; Grau, J.H.; Hofreiter, M.; Jeon, S.; Kliver, S.; Koepfli, K.P.; Meneghini, D.; Preick, M.; Tomarovsky, A; Totikov, A.; Fickel, J.; F√∂rster, D.W. Multiple types of genomic variation contribute to adaptive traits in the mustelid subfamily Guloninae. Molecular Ecology 2022. DOI: 10.1111/mec.16443.

## üí¨ Languages

**Russian**: Native <br>
**English**: Pre-Intermediate
<br><br>
